### ISSUE TITLE
Wan 2.2 Animate: Question about loss computation on noisy reference and temporal latents.

### ISSUE THREAD

**Issue by 0xLDF** [Wan-Video/Wan2.2#200]
Hi Wan team,
I’m trying to fine-tune Wan-2.2-Animate and would like to clarify how the loss is calculated for the two “extra” latents:
- noisy reference latents
- noisy temporal latents

From the inference code I see that the denoising network receives three kinds of noisy latents:
- noisy video latents
- noisy reference latents
- noisy temporal latents

The reference latents is also denoised step-by-step, so their value changes during sampling.
This implies that you must have computed a loss on both the reference and the temporal latent during training. Could you share the details?
Did you simply apply the same flow-matching loss to all three tensors and sum/average them?

Finally, I would like to ask: in current image/video generation models such as Flux-Kontext, Stand-in, Phantom, and RealisDance-DiT, the noisy latents never contain the reference image, and reference is not supervised by any loss during training. What was the motivation behind Wan-2.2-Animate’s decision to add noise to the reference latents.

**Comment by kamalmishra8825** [comment]
https://github.com/user-attachments/assets/a12bd8ff-305d-4dbd-89b3-a0ec5127f2ae

![Image](https://github.com/user-attachments/assets/da0e8801-c017-4859-95e6-f2cec19438b6)

**Comment by picolo100** [comment]
> FB_VID_7561383323959956345.mp4 
> ![Image](https://github.com/user-attachments/assets/da0e8801-c017-4859-95e6-f2cec19438b6)

?????????????????

**Comment by picolo100** [comment]
> Hi Wan team, I’m trying to fine-tune Wan-2.2-Animate and would like to clarify how the loss is calculated for the two “extra” latents:
> 
> * noisy reference latents
> * noisy temporal latents
> 
> From the inference code I see that the denoising network receives three kinds of noisy latents:
> 
> * noisy video latents
> * noisy reference latents
> * noisy temporal latents
> 
> The reference latents is also denoised step-by-step, so their value changes during sampling. This implies that you must have computed a loss on both the reference and the temporal latent during training. Could you share the details? Did you simply apply the same flow-matching loss to all three tensors and sum/average them?
> 
> Finally, I would like to ask: in current image/video generation models such as Flux-Kontext, Stand-in, Phantom, and RealisDance-DiT, the noisy latents never contain the reference image, and reference is not supervised by any loss during training. What was the motivation behind Wan-2.2-Animate’s decision to add noise to the reference latents.

interesting question i woudl also like to know the answer on this, have you got any further infos as you posted this on 09/25?
